--- ../hitch-1.4.7.original/src/hitch.c	2018-02-26 19:16:58.530326500 -0500
+++ ./src/hitch.c	2018-02-26 16:10:02.035227700 -0500
@@ -914,7 +914,6 @@
 	if (CONFIG->ALPN_PROTOS != NULL)
 		SSL_CTX_set_next_protos_advertised_cb(ctx, npn_select_cb, NULL);
 #endif
-	AN(SSL_CTX_set_default_verify_paths(ctx));
 
 	if (ciphers != NULL) {
 		if (SSL_CTX_set_cipher_list(ctx, ciphers) != 1) {
@@ -934,6 +933,10 @@
 	sc->staple_vfy = cf->ocsp_vfy;
 	VTAILQ_INIT(&sc->sni_list);
 
+	if (sc->staple_vfy > 0 ||
+	    (sc-> staple_vfy < 0 && CONFIG->OCSP_VFY))
+		AN(SSL_CTX_set_default_verify_paths(ctx));
+
 	if (CONFIG->PMODE == SSL_CLIENT)
 		return (sc);
 

