diff -ru ../hitch-1.4.7.original/configure ./configure
--- ../hitch-1.4.7.original/configure	2018-04-09 02:05:46.513731720 +0400
+++ ./configure	2018-04-16 04:35:46.566172104 +0400
@@ -580,8 +580,8 @@
 # Identity of this package.
 PACKAGE_NAME='hitch'
 PACKAGE_TARNAME='hitch'
-PACKAGE_VERSION='1.4.7'
-PACKAGE_STRING='hitch 1.4.7'
+PACKAGE_VERSION='1.4.7-1_6.cachewall'
+PACKAGE_STRING='hitch 1.4.7-1_6.cachewall'
 PACKAGE_BUGREPORT='opensource@varnish-software.com'
 PACKAGE_URL=''
 
@@ -4358,7 +4358,7 @@
 
 # Define the identity of the package.
  PACKAGE='hitch'
- VERSION='1.4.7'
+ VERSION='1.4.7-1_6.cachewall'
 
 
 cat >>confdefs.h <<_ACEOF
